portfolio/src/main/java/com/google/sps/data/ContactMeMessage.java

```diff
@@ -13,6 +18,32 @@ public ContactMeMessage(long id, long timestamp, String email, String message){
         this.message = message;
     }
 
+    public ContactMeMessage(long timestamp, String email, String message){
+        this.timestamp = timestamp;
+        this.email = email;
+        this.message = message;
+        this.id = 0;
+    }
+
+    public static ArrayList<ContactMeMessage> createList(QueryResults<Entity> results){
+        ArrayList<ContactMeMessage> list = new ArrayList<>();
+        
+        while(results.hasNext()){
+            list.add(entityConverter(results.next()));
+        }
+
+        return list;
+    }
+
+    public static ContactMeMessage entityConverter(Entity entity){
+        long id = entity.getKey().getId();
+        long timestamp = entity.getLong("timestamp");
+        String email = entity.getString("email");
+        String message = entity.getString("message");
+
+        return new ContactMeMessage(id, timestamp, email, message);
+    }
```

Comment:
Did you consider putting these methods in the DatastoreHelper class? Why did you decide to put them here instead?

Reply:
Yes, I considered it but since DatastoreHelper uses reflection I did not wanted to write methods for a specific class. I rather implement those in the class that is involved.

I did not wanted to use reflection for this because it involves the use of the specific constructor. I am sure there is a way to do it but as you said, it is sometimes too confusing.

Reply:
You could either wrap or extend DatastoreHelper<ContactMeMessage> with this behavior. If you use inheritance, maybe you would add to DatastoreHelper:

protected abstract T convertEntity(Entity entity);

Reply:
Great advise!
Done in the last commit